tools/Customization/DevHome.FileExplorerSourceControlIntegration/Services/RepositoryTracking.cs

```diff
@@ -150,6 +150,23 @@ public string GetSourceControlProviderForRootPath(string rootPath)
         }
     }
 
+    public void ModifySourceControlProviderForTrackedRepository(string extensionCLSID, string rootPath)
+    {
+        lock (trackRepoLock)
+        {
+            if (TrackedRepositories.ContainsKey(rootPath))

```

Comment:
NIT: Use .TryGet to prevent dipping into the dictionary twice.

Reply:
Modified in latest. Thanks!